warehouse/integrations/deltalake/deltalake.go

```diff
@@ -146,210 +149,202 @@ func NewDeltalake() *Deltalake {
 func stagingColumnNames(keys []string) string {
 	format := func(idx int, str string) string {
 		return fmt.Sprintf(`STAGING.%s`, str)
 	}
 	return warehouseutils.JoinWithFormatting(keys, format, ",")
 }
 
-// columnsWithValues returns main and staging column values.
 func columnsWithValues(keys []string) string {
 	format := func(idx int, str string) string {
 		return fmt.Sprintf(`MAIN.%[1]s = STAGING.%[1]s`, str)
 	}
 	return warehouseutils.JoinWithFormatting(keys, format, ",")
 }
 
-// checkAndIgnoreAlreadyExistError checks and ignores native errors.
-func checkAndIgnoreAlreadyExistError(errorCode, ignoreError string) bool {
-	if errorCode == "" || errorCode == ignoreError {
-		return true
-	}
-	return false
+func (*Deltalake) DeleteBy([]string, warehouseutils.DeleteByParams) error {
+	return fmt.Errorf(warehouseutils.NotImplementedErrorCode)
 }
 
-// NewClient creates deltalake client
-func (dl *Deltalake) NewClient(cred *client.Credentials, connectTimeout time.Duration) (Client *client.Client, err error) {
-	ctx := context.Background()
-	identifier := misc.FastUUID().String()
-	connConfig := &proto.ConnectionConfig{
-		Host:            cred.Host,
-		Port:            cred.Port,
-		HttpPath:        cred.Path,
-		Pwd:             cred.Token,
-		Schema:          dl.Schema,
-		SparkServerType: dl.SparkServerType,
-		AuthMech:        dl.AuthMech,
-		Uid:             dl.UID,
-		ThriftTransport: dl.ThriftTransport,
-		Ssl:             dl.SSL,
-		UserAgentEntry:  dl.UserAgent,
-	}
-
-	// Getting timeout context
-	timeout := dl.GrpcTimeout
-	if connectTimeout != 0 {
-		timeout = connectTimeout
-	}
-	tCtx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
-
-	// Creating grpc connection using timeout context
-	conn, err := grpc.DialContext(tCtx, dl.ConnectorURL, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
-	if err == context.DeadlineExceeded {
-		execTimeouts := dl.Stats.NewStat("warehouse.deltalake.grpcTimeouts", stats.CountType)
-		execTimeouts.Count(1)
-
-		err = fmt.Errorf("connection timed out to Delta lake: %w", err)
-		return
-	}
-	if err != nil {
-		err = fmt.Errorf("error while creating grpc connection to Delta lake: %w", err)
-		return
-	}
+func (d *Deltalake) fetchTables() ([]string, error) {
+	var (
+		query     string
+		startedAt time.Time
+		rows      *sql.Rows
+		err       error
+		tables    []string
+	)
 
-	dbClient := proto.NewDatabricksClient(conn)
-	connectionResponse, err := dbClient.Connect(ctx, &proto.ConnectRequest{
-		Config:     connConfig,
-		Identifier: identifier,
-	})
-	if err != nil {
-		err = fmt.Errorf("error connecting to Delta lake: %w", err)
-		return
-	}
-	if connectionResponse.GetErrorCode() != "" {
-		err = fmt.Errorf("error connecting to Delta lake with response: %s", connectionResponse.GetErrorMessage())
-		return
-	}
+	query = fmt.Sprintf(`SHOW tables from %s;`, d.Namespace)
+
+	defer func() {
+		d.Logger.Infow("fetching tables",
```

Comment:
It seems a bit too much to have these as INFO. I think it makes more sense to have them as DEBUG if you really need them and then have a metric for the query execution time.

Reply:
The same applies to similar defers in this code.

Reply:
Added slow query logs based on execution time.

Reply:
https://github.com/rudderlabs/rudder-server/pull/3190